di_website/visualisation/static/visualisation/widgets/js/ace-editor.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+const renderChart = (options, element) => {
+    const { data, layout } = JSON.parse(options);
+    return Plotly.newPlot(element, data, layout);
+};
+const initPlotlyPreview = (widgetID, options) => {
```

Comment:
@edwinmp 

This implies that editing of the exported JSON data can take place within the Wagtail page itself?

I'd be _very_ wary of allowing that, as you will likely end up with divergence, the danger being that an editor makes changes in Wagtail that are then overwritten by a new data export from Chart Studio.

Reply:
I believe allowing editing within Wagtail is important since we don't have a finite number of charts on Plotly Studio under the free accounts we intend to use. That means as more and more charts are added, we'll depend more on Wagtail if we want to edit the older charts. Also, it makes things easier for advanced users.
To that end, I'm using a code formatter to make Wagtail edits easy:
![image](https://user-images.githubusercontent.com/5672438/85101375-f058f400-b20a-11ea-9f3b-02469447483f.png)

Reply:
@k8hughes @benjamincoleman @edwinmp 

I understand that this feature may be useful, but my understanding of the process was that Wagtail was to be used as a data store for chart data, as opposed to an editing interface.

Here's a scenario in which I see a risk:

- User A creates a chart in Chart Studio, exports the data, and adds the data to a Wagtail chart page
- User B modifies the data in the Wagtail page using the JSON editor
- User A refines the chart in Chart Studio, re-exports the data, and updates the existing Wagtail page

In this instance, the modifications made by User B are lost as the chart no longer has a single source of truth (Chart Studio).

Just thought I'd flag this up here as this feels like this implementation is diverging a little from the process that's been discussed so far.